node/rpcstack.go

```diff
@@ -261,7 +266,12 @@ func (h *httpServer) doStop() {
 		h.wsHandler.Store((*rpcHandler)(nil))
 		wsHandler.server.Stop()
 	}
-	h.server.Shutdown(context.Background())
+	ctx, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
+	defer cancel()
+	err := h.server.Shutdown(ctx)
+	if err != nil {
+		h.log.Warn("Something wrong with HTTP server graceful shutdown", "error", err)
```

Comment:
I think we should call `h.server.Close` here to terminate the remaining connections non-gracefully.

Reply:
hmm fair point, I have updated the diff.
btw I found the `Shutdown()` returns `err` not only on context timeout but also on closing listeners fail, which makes no need to call `Close()`, so put a check `err` against `DeadlineExceeded`.